### PR TITLE
Changed replace NewStatus with Status

### PR DESCRIPTION
**Pull request by Seaven**
Issused #2000 replace NewStatus with Status

### CODE REVIEW

File: be/src/olap/utils.cpp [hunk 1]
```diff
@@ -1424,6 +1425,24 @@ int Errno::no() {
     return errno;
 }
 
+static Status disk_error(const std::string& context, int16_t err) {
```
Comment (imay) [inline be/src/olap/utils.cpp]: We have this code in env/env_posix.cpp, should reuse it other than coping it again.

File: be/src/olap/utils.cpp [hunk 2]
```diff
@@ -1432,9 +1451,10 @@ OLAPStatus dir_walk(const string& root,
     struct dirent* direntp = NULL;
     dirp = opendir(root.c_str());
     if (dirp == nullptr) {
-        NewStatus status = IOError(strings::Substitute("opendir $0 failed", root), errno);
-        LOG(WARNING) << status.ToString();
-        if (status.IsDiskFailure()) {
+        Status status = disk_error("opendir failed", errno);
```
Comment (imay) [inline be/src/olap/utils.cpp]: 1. After you replace NewStatus, it is not used by anyone. You should remove these files and modify NOTICE too.
2. You can reimplement this function with interface in env.

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/olap/utils.cpp (hunk 2 above)
Comment (Seaven) [reply be/src/olap/utils.cpp]: Yeah.
1. Will remove the new_status file
2. I will refactor the method in other issused

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**